pkg/fleet/internal/msi/msiexec.go

```diff
@@ -0,0 +1,281 @@
+			// So include a bit of context before and after
+			return FindAllIndexWithContext(regexp.MustCompile("Failed to CreateService"), bytes, 5, 5)
+		},
+		func(bytes []byte) [][]int {
+			// Typically looks like this:
+			//  Calling custom action AgentCustomActions!Datadog.AgentCustomActions.CustomActions.ProcessDdAgentUserCredentials
+			//	CA: 01:49:43: LookupAccountWithExtendedDomainSyntax. User not found, trying again with fixed domain part: \toto
+			//	CA: 01:49:43: ProcessDdAgentUserCredentials. User toto doesn't exist.
+			//	CA: 01:49:43: ProcessDdAgentUserCredentials. domain part is empty, using default
+			//	CA: 01:49:43: ProcessDdAgentUserCredentials. Installing with DDAGENTUSER_PROCESSED_NAME=toto and DDAGENTUSER_PROCESSED_DOMAIN=datadoghq-qa-labs.local
+			//	CA: 01:49:43: HandleProcessDdAgentUserCredentialsException. Error processing ddAgentUser credentials: Datadog.CustomActions.InvalidAgentUserConfigurationException: A password was not provided. A password is a required when installing on Domain Controllers.
+			//	   at Datadog.CustomActions.ProcessUserCustomActions.ProcessDdAgentUserCredentials(Boolean calledFromUIControl)
+			//	MSI (s) (C8!50) [01:49:43:906]: Product: Datadog Agent -- A password was not provided. A password is a required when installing on Domain Controllers.
+			//
+			//	A password was not provided. A password is a required when installing on Domain Controllers.
+			//	CustomAction ProcessDdAgentUserCredentials returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
+			//	Action ended 1:49:43: ProcessDdAgentUserCredentials. Return value 3.
+			// So include lots of context to ensure we get the full picture
+			return FindAllIndexWithContext(regexp.MustCompile("A password was not provided"), bytes, 6, 6)
+		},
+		func(bytes []byte) [][]int {
+			// Typically looks like this:
+			// 	Info 1603. The file C:\Program Files\Datadog\Datadog Agent\bin\agent\process-agent.exe is being held in use by the following process: Name: process-agent, Id: 4704, Window Title: '(not determined yet)'. Close that application and retry.
+			// Not much context to be had before and after
+			return FindAllIndexWithContext(regexp.MustCompile("is being held in use by the following process"), bytes, 1, 1)
+		},
+		func(bytes []byte) [][]int {
+			// Typically looks like this:
+			// 	Calling custom action AgentCustomActions!Datadog.AgentCustomActions.CustomActions.StartDDServices
+			// 	CustomAction WixFailWhenDeferred returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
+			// 	Action ended 2:11:49: InstallFinalize. Return value 3.
+			// The important context is the line after the error ("Return value 3") but the previous lines can include some useful information too
+			return FindAllIndexWithContext(regexp.MustCompile("returned actual error"), bytes, 5, 1)
+		})
```

Comment:
We should add the error when a service is pending delete, it's a misleading 'access denied" type error. See [wiki](https://datadoghq.atlassian.net/wiki/spaces/TS/pages/3111879791/Windows+Agent+6+7.47+Install+Log+Troubleshooting#Service-%3Cservice%3E-could-not-be-installed.-Verify-that-you-have-sufficient-privileges-to-install-system-services.) for a snippet.